.github/workflows/e2ev2-matrix.yaml

```diff
@@ -0,0 +1,54 @@
+name: E2E Version 2 Test Matrix
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        type: string
+      skipRefCheck:
+        type: boolean
+        default: true
+
+permissions:
+    id-token: write
+
+jobs:
+  plan-infra:
+    runs-on: ubuntu-latest
+    outputs:
+       matrix: ${{ steps.matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '~1.20.3'
+          cache-dependency-path: "**/*.sum"
+
+      - run: |
+          cd testing/e2e
+          go run ./main.go matrix
+        shell: bash
+        id: matrix
+        if:
```

Comment:
i also see some formatting strangeness here
![image](https://github.com/Azure/aks-app-routing-operator/assets/4248857/411099db-c7b8-431e-b5b3-d29e04766558)

Reply:
what is strange about this? nothing looks weird to me

Reply:
it's been tested and working fine. wouldn't worry about weird syntax

Reply:
the keys are the same color as the value - but if it works fine it's probably a github ui glitch

Reply:
github sev 1